### DISCUSSION TITLE
Seeking Advice on Versioning Strategy

### DISCUSSION THREAD

**Discussion by MDeLuise** [MDeLuise/plant-it#175] [General]
Hi everyone,
I'm opening this discussion in order to  get some ideas about handling the versioning strategy for the project.

The project consists of a single Docker container running two services: a server and a web app. Initially, these services were split into two separate images, but to simplify the deployment process, they were merged them into one container.
We're approaching the release of the "first official" version, `1.0.0`, which will include both the web app and the server, each marked as version `1.0.0`.

I'm seeking advice on how to handle the versioning system moving forward. If the web app and server were in separate repositories, versioning would be straightforward: each would follow semantic versioning independently, and the repository version would match the respective service version.

However, with both services in the same repository, versioning becomes more complex. I’ve done some research to see how other projects manage this, such as a discussion on the Immich project (see [here](https://github.com/immich-app/immich/discussions/2636)). They said that they are still in a fast development cycle and thus use versioning to track progress, planning for more strict versioning rules upon reaching a stable release. So not really something useful for now.

Other related discussions on Stack Overflow provide insights too:
- [Best practices for managing versions between backend and frontend](https://stackoverflow.com/questions/69912061/best-practices-for-managing-versions-between-backend-and-frontend)
- [Is it a good idea to keep separate projects in the same repository if they are interdependent?](https://stackoverflow.com/questions/8937728/is-it-a-good-idea-to-keep-separate-projects-in-the-same-repository-if-they-are-i)
- [How do you handle versioning in a multi-sided project?](https://softwareengineering.stackexchange.com/questions/264238/how-do-you-handle-versioning-in-a-multi-sided-project)

One straightforward solution is to update both components synchronously. For example, if a new release (`1.0.1`) includes a fix only for the web app, both the web app and server would be versioned as `1.0.1` (even though the server version `1.0.1` is identical to `1.0.0`). 

**Pros**:
- Simplifies the overall project versioning.
- Keeps the GitHub repository version in sync with the components.

**Cons**:
- It's not entirely accurate since the server versions would be identical despite the version increment.

I believe this might be the preferred approach, but I’m open to suggestions. If you have any other ideas or best practices for handling this kind of versioning in a combined repository setup, please share them below. Your insights will be greatly appreciated!

**Comment by KickingAss2024** [comment]
I don't know what's best. Would an all-in-one solution be easier to get approval to be added to TrueCharts, as it will run on Kubernetes?

Is this something your thinking of doing?
